lib/SILOptimizer/Transforms/DeadObjectElimination.cpp

```diff
@@ -204,6 +205,23 @@ static bool canZapInstruction(SILInstruction *Inst, bool acceptRefCountInsts) {
       isa<DeallocPartialRefInst>(Inst))
     return acceptRefCountInsts;
 
+  if (isa<InjectEnumAddrInst>(Inst))
+    return true;
+
+  // We know that the destructor has no side effects so we can remove the
+  // deallocation instruction too.
+  if (isa<DeallocationInst>(Inst) || isa<AllocationInst>(Inst))
+    return true;
+
+  // Much like deallocation, destroy addr is safe.
+  if (isa<DestroyAddrInst>(Inst))
+    return true;
+
+  // The only store instructions which is guaranteed to store a trivial value
+  // is an inject_enum_addr witout a payload (i.e. without init_enum_data_addr).
+  if (onlyAcceptTrivialStores)
```

Comment:
I find onlyAcceptTrivialStores to be confusing slightly. I think what you are really trying to say that the value is non-trivial, so only inject_enum_addr without a payload is the only way to have a trivial store. Can you make that clearer? I had to think about this to make sure it made sense to me. That says to me that it needs a comment or to be written clearer.

Reply:
That's what I already wrote in that comment. If you have a specific suggestion for a better comment, please let me know

Reply:
I would add into the comment that this applies only to non-trivial types. The reason I was confused is b/c you can store [trivial]. That is why it was confusing to me.